src/renderer/components/+extensions/install.tsx

```diff
@@ -30,17 +28,22 @@ interface Dependencies {
   extensionInstallationStateStore: ExtensionInstallationStateStore;
 }
 
-const installInputValidators = [
-  InputValidators.isUrl,
-  InputValidators.isPath,
-  InputValidators.isExtensionNameInstall,
-];
```

Comment:
The problem is that the old code was had a bug in it. It would cause a "UnhandledPromiseRejectionError" because `isPath` is an async validator

Reply:
Well, I still think this new function `installInputValidator` should be simplified or splitted to smaller ones. If we want to add some new validator to the list, we will touch code of this `installInputValidator` and probably add some another ifs or replace conditions which is not good. Not good because it can break other things.

Maybe we can use `for ... of` to iterate through the validators array https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of

My quick checks shows that it "handles" async and non-async items:

```
const x = () => true
const y = () => true
const z = () => {
    return new Promise(resolve => {
        resolve(true);
    });
}

for await (const variable of [x, y, z]) {
    console.log(variable())
}

// prints
// VM593:2 true
// VM593:2 Promise {<fulfilled>: true}
```

Reply:
The problem is that in the async case, it isn't that the promise should resolve to `true` but that it shouldn't reject. This is a hold over from how they were previously implemented.

Reply:
Haven't read everything in this PR.. but just quick question:
if we don't want to that some async validator _might throw exception_ why we cannot just wrap it with extra `.catch(() => true)` or something?

Reply:
If I'm not mistaken if we wrap every promise with extra `.catch(() => {})` we can filter all relevant errors / skip   `UnhandledPromiseRejectionError` or if it's eligeble or just passthrough inside catch with `throw` again.

e.g. something like this wouldn't work?

```ts
// validation cycle
await Promise.resolve(validator()).catch(err => {
   if(err instanceof UnhandledPromiseRejectionError) return true; // all good
   throw String(err); // pass real validation error
})
```

Reply:
That seems very hacky and wrong, especially since that is global shared state. It isn't that hard to make a union function that works correctly. Just shows the horribleness of the current design.